### PR TITLE
Expand comma separated platform names correctly

### PR DESCRIPTION
**Pull request by wxtim**
This is a small change with no associated Issue.

## Issue summary:

#4854 did not expand comma separated platform config items in the dense config.

**Requirements check-list**
- [x] I have read `CONTRIBUTING.md` and added my name as a Code Contributor.
- [x] Contains logically grouped changes (else tidy your branch by rebase).
- [x] Does not contain off-topic changes (use other PRs for other changes).
- [x] Applied any dependency changes to both `setup.cfg` and `conda-environment.yml`.
- [x] Appropriate tests are included (unit and/or functional).
- [x] No change log entry required (direct follow on to #4860 ).
- [x] No documentation update required.

### CODE REVIEW

File: cylc/flow/cfgspec/globalcfg.py [hunk 1]
```diff
@@ -1493,9 +1493,11 @@ def load(self):
                     LOG.error(f'bad {conf_type} {fname}')
                     raise
 
+        # Expand platforms needs to be performed fist because it
```
Comment (datamel) [inline cylc/flow/cfgspec/globalcfg.py:1496]:
```suggestion
        # Expand platforms needs to be performed first because it
```

**Comment by MetRonnie** [comment]
> #4860 did not expand comma separated platform config items in the dense config.

Wrong issue?

**Comment by wxtim** [comment]
> Wrong PR? 

Yes #4854 

> And isn't the problem that it didn't expand in the sparse config?

The problem is that it expands the sparse config _after_ creating the dense config, and therefore doesn't include that expansion in the dense config.

File: cylc/flow/cfgspec/globalcfg.py [hunk 2]
```diff
@@ -1493,9 +1493,11 @@ def load(self):
                     LOG.error(f'bad {conf_type} {fname}')
                     raise
 
+        # Expand platforms needs to be performed first because it
+        # manipulates the sparse config.
+        self._expand_platforms()
         self._set_default_editors()
```
Comment (MetRonnie) [inline cylc/flow/cfgspec/globalcfg.py:1499]: I think the fact that the dense config is expanded in a method called `_set_default_editors()` is a bit troublesome. I am addressing this as part of #4815

**Review by MetRonnie — APPROVED** [review]

**Review by datamel — APPROVED** [review]
Checked out and tested. Thanks @wxtim.

**Pull request merged**